code/controllers/subsystem/security_level.dm

```diff
@@ -56,6 +56,13 @@ SUBSYSTEM_DEF(security_level)
 		SSshuttle.emergency.modTimer(old_shuttle_call_time_mod)
 		SSshuttle.emergency.modTimer(selected_level.shuttle_call_time_mod)
 
+	//SKYRAT EDIT ADDITION START- In the event of an orange alert, change the airlock permissions
+	if (istype(current_security_level, /datum/security_level/orange))
```

Comment:
this is not what I meant when I said to make it use signals

please make doors that you want to be emergency opened by engineers register a proc to the signal of the alert level changing, and then have them adjust their status based on that